scrapy_poet/utils/__init__.py

```diff
@@ -37,6 +42,7 @@ def http_request_to_scrapy_request(request: HttpRequest, **kwargs) -> Request:
         method=request.method,
         headers=request.headers,
         body=request.body,
+        callback=NO_CALLBACK,
```

Comment:
I'm not sure about this approach since there's some parts of the code that checks for the value of `request.callback` if it's `None`.

What about updating this part instead which reduces the scope to the part where we build the dependencies for a given callback? https://github.com/scrapinghub/scrapy-poet/blob/master/scrapy_poet/injection.py#L344-L345

Reply:
What would you change in that function? Can it assume that `request.callback is None` means no callback?

Reply:
Yes, my thoughts exactly

```python
if request.callback is None:
    return NO_CALLBACK
```

Reply:
Is it safe to change this?

Reply:
Probably safer in the context of this fix since the `Injector` is the only one using it at the moment.

Reply:
This one in particular https://github.com/scrapinghub/scrapy-poet/blob/master/scrapy_poet/injection.py#L140-L151 causes https://github.com/scrapy-plugins/scrapy-zyte-api/issues/135 because of `parse`.

Reply:
I just wonder if it's possible to get a request there that has `callback=None` but actually wants `self.parse`. Like one from `start_urls`?

Reply:
Yeah, it's indeed executed for requests from `start_urls` and so deps for those aren't created.

Reply:
Ahhh that's a good point! It seems your original approach would be better here. 👍